src/SecureHeaders.php

```diff
@@ -875,6 +923,43 @@ private function pipeline()
             );
         }
 
+        if (
+            ($this->automaticHeaders & self::AUTO_COOKIE_SAMESITE)
+            === self::AUTO_COOKIE_SAMESITE
+            and (
+                ! isset($this->sameSiteCookies)
+                or $this->sameSiteCookies === 'Lax'
+                or $this->sameSiteCookies === 'Strict'
+            )
+        ) {
+            # add SameSite to cookies that look like they hold
+            # session data
+
+            if ( ! isset($this->sameSiteCookies) and $this->strictMode)
+            {
+                $sameSite = 'Strict';
+            }
+            elseif ( ! isset($this->sameSiteCookies))
+            {
+                $sameSite = 'Lax';
+            }
+            else
+            {
+                $sameSite = $this->sameSiteCookies;
+            }
```

Comment:
Can you extract both this if/elseif/else block and the second part of the large if condition above it into methods with meaningful names?